### PR TITLE
Fix assert when a function been called more than once by entry.

### PR DESCRIPTION
**Pull request by python3kgae**
For better support of noinline.

If a function is processed, just skip in DxilViewIdStateBuilder::ComputeReachableFunctionsRec.

Added ViewIdStatePrinter pass to test the change.

### CODE REVIEW

File: lib/HLSL/ComputeViewIdStateBuilder.cpp [hunk 1]
```diff
@@ -925,3 +927,47 @@ ModulePass *createComputeViewIdStatePass() {
 }
 
 } // end of namespace llvm
+
+namespace {
+class ViewIdStatePrinter : public ModulePass {
```
Comment (pow2clk) [inline lib/HLSL/ComputeViewIdStateBuilder.cpp:932]: I'm not very familiar with these printer passes, but it looks to me that we have very few of them explicitly specified. Instead, if you add a `print()` method to the `ComputeViewIdState` pass and invoke `opt` with `-viewid-state -analyze`, I think the `opt.cpp` code will invoke your print function using the `ModulePassPrinter` class

**Review by pow2clk — COMMENTED** [review]
It looks good, but I fear we might be going against the "usual way" of doing things. I hate to trip you up, but I think it's just a matter of moving the code you've written, which looks fine into a different function of the ComputeViewIdState pass

File: lib/HLSL/ComputeViewIdStateBuilder.cpp (hunk 1 above)
Comment (python3kgae) [reply lib/HLSL/ComputeViewIdStateBuilder.cpp]: Never know that. New things learned.
Fixed.

**Review by pow2clk — APPROVED** [review]
Looks good!

**Comment by llvm-beanz** [comment]
The tests in this PR only run under lit. We cannot add tests that only run under lit until we remove support for the non-LIT based test workflows.

@python3kgae, can you please convert these to non-lit tests?

**Pull request merged**